src/libstd/keyword_docs.rs

```diff
@@ -925,8 +925,42 @@ mod mut_keyword {}
 //
 /// Make an item visible to others.
 ///
-/// The documentation for this keyword is [not yet complete]. Pull requests welcome!
+///`pub` makes any module, function, or data structure accessible from inside
+/// of external modules.
+/// ```rust
+/// mod other_mod {
+///     pub mod public_module {
+///         pub struct PublicStruct;
+/// }
+///     let emptyStruct = other_mod::public_module::PublicStruct;
+/// ```
+/// private_module does not use the `pub` keyword and therefore
+/// HiddenStruct, despite it's usage of the `pub` is inaccessible.
+///
+/// ```rust
```

Comment:
We should make this test case as `rust,compile_fail` and include the compilation error uncommented.